_posts/2017-05-10-april.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: post
+title: "Dev Diary (April 2017)"
+author: "Corey Montella"
+tags: []
+---
+
+_(Eve is a new programming language, and this is our development blog. If you’re new to Eve, [start here](http://witheve.com))_
+
+In April we continued our effort to release v0.3 of Eve. On April 18th we released a second preview of the v0.3 platform, which included several long-requested features. However, toward the end of the month we shifted focus to revamping documentation and making the "getting started" experience for Eve more accessible to new users. Let's take a look at each of these efforts in this month's diary.
+
+### Platform
+
+We released a second preview of the v0.3 runtime, featuring multiple parent roots, which enable you to embed Eve-rendered elements into an application; and streamlined watchers with built in support for more DOM events and triggers in the HTML watcher. These are covered in detail in the release notes, but I'll take a moment now to highlight how multiple parent roots work.
+
+#### Embedding Eve-rendered Elements in Your Application
+
+We've received some requests from people who would like to use Eve to render specific elements in an already existing project. To date, you could only parent Eve-rendered elements to the document root. With v0.3, you'll be able to embed any piece of Eve-rendered UI into your application arbitrarily. First, you add a reference to the parent element into Eve:
+
+```javascript
+import {Program} from "witheve";
+
+let program = new Program("my program");
+let htmlWatcher = program.attach("html");
+
+let someElement = document.querySelector("#eve-wrapper");
+htmlWatcher.addExternalRoot("my-root", someElement);
+``` 
+
+From within Eve, you can search for that element and add children to it, as you would any other record:
+
+```eve
+search
+  root = [#my-root]
+
+bind
+  root.children += [#ui/text text: "Hello!"]
+```
+
+Eve behaves as expected, updating the embedded element as records change. For new users, this is a nice way to introduce Eve to an existing project; instead of porting an application from scratch, you can dip your toes in the water and build a new component in Eve, then inject it into your project wherever you want. As you get more comfortable with Eve, you can use it in more of you application.
```

Comment:
This sections is 👍 Without really knowing much about Eve, I can get my head around what's going on here.